Tools/CISupport/ews-build/master.cfg

```diff
@@ -60,16 +60,21 @@ if not is_test_mode_enabled:
         },
     )
 
-    credentials = load_password('EWS_CREDENTIALS')
-    if not credentials:
-        print('EWS credentials not found. Please ensure EWS_CREDENTIALS is configured either in env variables or in passwords.json')
+    GITHUB_CLIENT_ID = load_password('GITHUB_CLIENT_ID')
+    GITHUB_CLIENT_SECRET = load_password('GITHUB_CLIENT_SECRET')
+    if (not GITHUB_CLIENT_ID) or (not GITHUB_CLIENT_SECRET):
+        print('EWS credentials not found. Please ensure GITHUB_CLIENT_ID and GITHUB_CLIENT_SECRET are configured either in env variables or in passwords.json')
         sys.exit(1)
+
     # See https://docs.buildbot.net/current/manual/configuration/www.html#example-configs
     authz = util.Authz(
-        allowRules=[util.AnyControlEndpointMatcher(role="admin")],
-        roleMatchers=[util.RolesFromEmails(admin=list(credentials.keys()))]
+        allowRules=[util.AnyEndpointMatcher(role='Administrators', defaultDeny=False),
+                    util.AnyEndpointMatcher(role='Bot Watchers', defaultDeny=False),
+                    util.RebuildBuildEndpointMatcher(role='Contributors'),
+                    util.AnyControlEndpointMatcher(role='Administrators')],
+        roleMatchers=[util.RolesFromGroups(groupPrefix='WebKit/')]
     )
-    auth = util.UserPasswordAuth(credentials)
+    auth = util.GitHubAuth(GITHUB_CLIENT_ID, GITHUB_CLIENT_SECRET, apiVersion=4, getTeamsMembership=True, debug=True)
```

Comment:
Typically, I would imagine debug mode to be insecure in at least some ways (like dumping information that could be useful to an attacker). Is this OK to deploy with it enabled?

Reply:
I checked it again. This one seems to be fine. It prints some extra information in the logs on the server, which might be helpful in debugging any issues (e.g: if users complaint that they didn't get the required access), especially in first few days.